lib/wallet/src/Cardano/Wallet/Write/Tx/Balance.hs

```diff
@@ -473,7 +515,11 @@ balanceTransactionWithSelectionStrategyAndNoZeroAdaAdjustment
         let
             transform
                 :: Selection
-                -> ([(W.TxIn, W.TxOut)], [(W.TxIn, W.TxOut)], [W.TxOut], s)
+                -> ( [(W.TxIn, W.TxOut)]
+                   , [(W.TxIn, W.TxOut)]
+                   , [W.TxOut]
+                   , changeState
+                   )
```

Comment:
Perhaps we would benefit from making this a record type. 😄 (Though obviously not an issue with this particular PR.)

Reply:
Indeed — I think there's an opportunity to come up with a neat `selectAssets` / `selectCoins` abstraction as boundary between `balanceTransaction` and the underlying coin-selection!

Reply:
> Indeed — I think there's an opportunity to come up with a neat `selectAssets` / `selectCoins` abstraction as boundary between `balanceTransaction` and the underlying coin-selection!

Indeed. Once `balanceTx` is the sole consumer of coin selection, it opens up all kinds of internal refactoring/simplification opportunities!